modules/compute/gke-node-pool/variables.tf

```diff
@@ -433,3 +435,9 @@ variable "enable_private_nodes" {
   type        = bool
   default     = false
 }
+
+variable "num_node_pools" {
```

Comment:
Question: Do we want to have a separate variable `num_slices` for TPUs?

If the architectures of GPUs and TPUs are really different, we would at some point have to refactor things.

Reply:
`num_node_pools` in being used in place of `num_slices`.
I will share a design doc (internally) with information on supporting TPUs.

Reply:
I would prefer we use `num_slices` for TPUs and `num_node_pools` for GPUs when defining the interface that customers use. Its more consistent and alignd to how TPUs are defined. 
We can have logic in the implementation to set both these fields to be equal. Going through a migration exercise later would involve significant work.